File: Divide and Conquer/codeforces_1915F.cpp

// Problem link ===>> https://codeforces.com/problemset/problem/1915/F
// submission link ===>> https://codeforces.com/contest/1915/submission/241941449

// #include <iostream>
// #include <vector>
// #include <algorithm>

// using namespace std;

// int main()
// {
//     int t;
//     cin >> t;

//     while (t--)
//     {
//         int n;
//         cin >> n;

//         vector<pair<int, int>> abArray;
//         vector<int> arrayTwo;
//         for (int i = 0; i < n; i++)
//         {
//             int a, b;
//             cin >> a >> b;
//             abArray.push_back({a, b});
//             arrayTwo.push_back(b); // Store y-coordinates separately
//         }

//         sort(abArray.begin(), abArray.end());
//         sort(arrayTwo.begin(), arrayTwo.end()); // Sort y-coordinates

//         long long ans = 0;

//         int j = 0; // Index for arrayTwo
//         for (auto &ab : abArray)
//         {
//             int x = ab.second;

//             int count = 0;
//             while (j < n && arrayTwo[j] < x)
//             {
//                 count++;
//                 j++;
//             }

//             ans += count;
//         }

//         cout << ans << '\n';
//     }

//     return 0;
// }

#include <bits/stdc++.h>
using namespace std;
using ll = long long;
#include <ext/pb_ds/assoc_container.hpp>
#include <ext/pb_ds/tree_policy.hpp>
using namespace __gnu_pbds;
typedef tree<int, null_type, less<int>, rb_tree_tag, tree_order_statistics_node_update> pbds;
int32_t main()
{
    ios_base::sync_with_stdio(0);
    cin.tie(0);
    int t;
    cin >> t;
    while (t--)
    {
        int n;
        cin >> n;
        vector<pair<int, int>> v;
        pbds s;
        int x, y;
        for (int i = 0; i < n; i++)
        {
            cin >> x >> y;
            s.insert(y);
            v.push_back({x, y});
        }
        sort(v.begin(), v.end());
        long long ans = 0;
        for (auto u : v)
        {
            int x = u.second;
            int y = s.order_of_key(x);
            ans += y;
            s.erase(x);
        }
        cout << ans << '\n';
    }
    return 0;
}

File: Divide and Conquer/codeforces_1915F.py

# Problem link ===>> https://codeforces.com/problemset/problem/1915/F

# def main():
#     t = int(input())

#     while t > 0:
#         t -= 1
#         n = int(input())

#         ab_array = []
#         array_two = []
#         for i in range(n):
#             a, b = map(int, input().split())
#             ab_array.append((a, b))
#             array_two.append(b)

#         ab_array.sort()
#         array_two.sort()

#         ans = 0
#         j = 0
#         for a, b in ab_array:
#             while j < n and array_two[j] < b:
#                 count += 1
#                 j += 1
#             ans += count

#         print(ans)

# if __name__ == "__main__":
#     main()

import bisect


def main():
    t = int(input())

    while t > 0:
        t -= 1
        n = int(input())

        v = []  # List to store pairs (a, b)
        y_coordinates = []  # List to store y-coordinates for counting
        for _ in range(n):
            a, b = map(int, input().split())
            v.append((a, b))
            y_coordinates.append(b)  # Add to y_coordinates initially

        v.sort()  # Sort based on the first element (a)

        ans = 0
        for a, b in v:
            # Find insertion index efficiently
            index = bisect.bisect_left(y_coordinates, b)
            ans += index
            y_coordinates.pop(index)  # Remove the found element

        print(ans)


if __name__ == "__main__":
    main()
